### DISCUSSION TITLE
Setting compiler flags

### DISCUSSION THREAD

**Discussion by KenzieMac130** [jrouwe/JoltPhysics#663] [Q&A] [answered]
Sooo this might just be me being dumb but I have been trying to figure out how to set the "JPH" flags when compiling the project and how to get it to match in my application. Normally these would be options in a cmake config that then would be fed to `configure_file` which would generate a pre-compiled header in the build directory (which I would then copy to my projects relevant include directory) but it seems that it is actually ending up as direct compiler flags? I don't want to edit the vsxproj's directly since that gets wiped out. I am not using the fetch content method, nor am I including the cmake as a subproject since I want all my deps pre-built and ready to download the artifacts and link.

What is the proper way to set them or see their values in the world of cmake?

**Comment by jrouwe** [comment]
This is demonstrated in the [JoltPhysicsHelloWorld](https://github.com/jrouwe/JoltPhysicsHelloWorld) example, just toggle these flags:

https://github.com/jrouwe/JoltPhysicsHelloWorld/blob/d8570eb6dff0c2318d30996daa338d023f71fe5e/Build/CMakeLists.txt#L11-L41

**Comment by jrouwe** [comment, marked as answer]
Oh, I'm not reading it right, you're not using cmake fetch content / using a sub project. In that case you will need to replicate the options in your own cmake project, e.g.:

target_compile_definitions(YourProjectName PUBLIC JPH_DOUBLE_PRECISION)
